nitro-root-enclave-server/src/main.rs

```diff
@@ -0,0 +1,328 @@
+};
+
+// Maximum number of outstanding connections in the socket's
+// listen queue
+const BACKLOG: usize = 128;
+
+const NITRO_ROOT_ENCLAVE_EIF_PATH: &str = "/home/ec2-user/nitro_root_enclave.eif";
+const INBOUND_PORT: u16 = 9090;
+
+#[derive(Debug, Error)]
+pub enum NitroServerError {
+    #[error(display = "NitroServer: Non Success Status")]
+    NonSuccessStatus,
+    #[error(display = "NitroServer: InvalidRootEnclaveMessage")]
+    InvalidRootEnclaveMessage,
+    #[error(display = "NitroServer: Bincode Error:{:?}", _0)]
+    Bincode(#[error(source)] bincode::ErrorKind),
+    #[error(display = "NitroServer: Enclave error:{:?}", _0)]
+    EnclaveError(#[error(source)] NitroError),
+    #[error(display = "NitroServer: Hex error:{:?}", _0)]
+    HexError(#[error(source)] hex::FromHexError),
+    #[error(display = "NitroServer: Base64 Decode error:{:?}", _0)]
+    Base64Decode(#[error(source)] base64::DecodeError),
+    #[error(display = "NitroServer: Invalid Protocol Buffer Message")]
+    InvalidProtoBufMessage,
+    #[error(display = "NitroServer: Non-Success HTTP Response received")]
+    NonSuccessHttp,
+    #[error(display = "NitroServer: Colima error:{:?}", _0)]
+    Colima(colima::custom::ColimaError),
+    #[error(display = "NitroServer: Curl error:{:?}", _0)]
+    Curl(curl::Error),
+    #[error(display = "NitroServer: Unimplemented")]
+    UnimplementedError,
+}
+
+fn main() {
+    println!("Hello, world!");
+    let matches = App::new("nitro-root-enclave-server")
+        .arg(Arg::with_name("tabasco")
+            .takes_value(true)
+            .help("URL for Tabasco server"))
+        .arg(
+            Arg::with_name("debug")
+                .long("debug")
+                .takes_value(false)
+                .help("Enables debug mode in the enclave")
+        )
+        .get_matches();
+    let tabasco_url = matches.value_of("tabasco").unwrap();
+    let enclave_debug = matches.is_present("debug");
+
+    // first, start the nitro-root-enclave
+    let enclave = loop {
+        match native_attestation(tabasco_url, enclave_debug) {
+            Err(err) => {
+                println!("nitro-root-enclave-server::main native_attestation failed({:?}). Sleeping and trying again.", err);
+                std::thread::sleep(std::time::Duration::from_secs(2));
+                continue;
+            },
+            Ok(enc) => break enc,
+        }
+    };
+
+    let socket_fd = socket(
+        AddressFamily::Inet,
+        SockType::Stream,
+        SockFlag::empty(),
+        None,
+    )
+    .expect("Failed to create socket_td");
+
+    let ip_string = local_ipaddress::get()
+        .expect("Failed to get local ip address");
+    let ip_addr: Vec<u8> = ip_string.split(".")
+        .map(|s| s.parse().expect("Parse error")).collect();
+    let my_ip_address = InetAddr::new(IpAddr::new_v4(ip_addr[0], ip_addr[1], ip_addr[2], ip_addr[3]), INBOUND_PORT);
```

Comment:
Is there any sanity check that `ip_addr` has a length >= 4 before this?

Reply:
I understand that abstractly we need to do a sanity check, but in reality, when would `local_ipaddress::get()` not fail and return a malformed ip address?

Reply:
Ah OK, I didn't notice that `local_ipaddress::get()` is a dedicated method for getting an IP address–––I thought this was string splitting.  I agree that there's no real need to check this, then.